web/src/main/java/org/cbioportal/web/parameter/DataBinFilter.java

```diff
@@ -5,11 +5,28 @@
 import java.util.List;
 
 import javax.validation.constraints.AssertTrue;
+import org.springframework.validation.annotation.Validated;
 
+@Validated
 public class DataBinFilter implements Serializable {
 
+    public enum BinMethod {
+        MEDIAN,
+        QUARTILE,
+        CUSTOM,
+        GENERATE;
```

Comment:
Should we name this method `ANCHORED` or something like that to make it more clear?

Reply:
No, the _generate_ approach takes two params the _anchor_ and the _bin size_. So `ANCHORED` does not sufficiently describe it.

Reply:
For the `MEDIAN` and `QUARTILE` methods the bins are also generated, right? So we generate bins for all methods other than `CUSTOM`. At least that is how I see it. If you think `GENERATE` describes this method the best I am okay with that though.

Reply:
These names reflect the UI elements in the frontend that control them. Would it not be a good idea to match these?